Library/Homebrew/cask/lib/hbc/cask_loader.rb

```diff
@@ -31,6 +31,12 @@ def initialize(path)
 
         @token = path.basename(".rb").to_s
         @path = path
+
+        @tap ||= if @path == CaskLoader.default_path(@token)
```

Comment:
Unfortunately, this would make `--full-name` not work, since the locally-installed casks are not loaded with `FromTapLoader`.

The two conditionals use full paths (`default_path` resolves to something like `/usr/local/Homebrew/Library/Taps/caskroom/homebrew-cask/Casks/foo.rb`, and `HOMEBREW_TAP_PATH_REGEX` to something like `(?-mix:\\/usr\\/local\\/Homebrew\\/Library\\/Taps\\/([\\w-]+)\\/([\\w-]+)\\/(.*))`), so I believe we can be quite confident that this is the correct tap. This is the same logic that is used in `Formula#initialize`, which is why I originally felt comfortable putting it directly in `Hbc::Cask#initialize` instead of `FromPathLoader`.